# Start here for serious study

- MIT Open Courseware's [Foundations of Computational And-Systems Biology, Spring 2014](https://ocw.mit.edu/courses/biology/7-91j-foundations-of-computational-and-systems-biology-spring-2014/index.htm). 
This course is an introduction to computational biology emphasizing the fundamentals of nucleic acid and protein sequence and structural analysis; it also includes an introduction to the analysis of complex biological systems. Topics covered in the course include principles and methods used for sequence alignment, motif finding, structural modeling, structure prediction and network modeling, as well as currently emerging research areas. [Video Lectures from Spring 2014 at MIT](https://ocw.mit.edu/courses/biology/7-91j-foundations-of-computational-and-systems-biology-spring-2014/video-lectures/).

## Coursera

Bioinformatics as taught by Pavel Pevzner and Phillip Compeau of California San Diego.

- [Bioinformatics I: Finding Hidden Messages in DNA](https://www.coursera.org/learn/dna-analysis): In the first half of the course, we investigate DNA replication, and ask the question, where in the genome does DNA replication begin?  We will see that we can answer this question for many bacteria using only some straightforward algorithms to look for hidden messages in the genome. In the second half of the course, we examine a different biological question, when we ask which DNA patterns play the role of molecular clocks.  The cells in your body manage to maintain a circadian rhythm, but how is this achieved on the level of DNA? 

- [Bioinformatics II: Genome Sequencing](https://www.coursera.org/learn/genome-sequencing): Biologists still cannot read the nucleotides of an entire genome as you would read a book from beginning to end. However, they can read short pieces of DNA. In this course, we will see how graph theory can be used to assemble genomes from these short pieces. We will further learn about brute force algorithms and apply them to sequencing mini-proteins called antibiotics. 

- [Bioinformatics III: Comparing Genes, Proteins, and Genomes](https://www.coursera.org/learn/comparing-genomes):Once we have sequenced genomes in the previous course, we would like to compare them to determine how species have evolved and what makes them different.

- [Bioinformatics IV:  Molecular Evolution](https://www.coursera.org/learn/molecular-evolution): One way we can use these methods is in order to construct a "Tree of Life" showing how a large collection of related organisms have evolved over time.

- [Bioinformatics V: Genomic Data Science and Clustering](https://www.coursera.org/learn/genomic-data): How do we infer which genes orchestrate various processes in the cell?  How did humans migrate out of Africa and spread around the world? In this class, we will see that these two seemingly different questions can be addressed using similar algorithmic and machine learning techniques arising from the general problem of dividing data points into distinct clusters.

- [Bioinformatics VI: Finding Mutations in DNA and Proteins](https://www.coursera.org/learn/dna-mutations): This course will cover advanced topics in finding mutations lurking within DNA and proteins. The approach we will use is based on a powerful machine learning tool called a hidden Markov model.

### Other online courses

- [DIY Transcriptomics](http://diytranscriptomics.com/): A virtual asynchronous course covering best practices for RNAseq data analysis, uses R/bioconductor environment.

- [RNA-Seq with Bioconductor in R](https://www.datacamp.com/courses/rna-seq-with-bioconductor-in-r): We will start the course with a brief overview of the RNA-Seq workflow with an emphasis on [differential expression (DE) analysis](https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html).  The DESeq2 R package will be used to model the count data using a negative binomial model and test for differentially expressed genes.